docs/references/strategies/languages/python/pyproject-generic.md

```diff
@@ -0,0 +1,62 @@
+# PyProject.toml Generic Support
+
+The FOSSA CLI provides robust support for analyzing Python projects that use the [PEP 621](https://peps.python.org/pep-0621/) `pyproject.toml` format, which is becoming the standard for Python project metadata and configuration.
+
+## Project Discovery
+
+The CLI searches for `pyproject.toml` files and automatically detects which build system is being used by examining the file contents.
+
+## Project Type Detection
+
+When analyzing a `pyproject.toml` file, the CLI will attempt to determine which build system is in use by checking for specific sections:
+
+1. **Poetry**: Detected by the presence of `[tool.poetry]` section
+2. **PDM**: Detected by the presence of `[tool.pdm]` section
+3. **PEP 621**: Detected by the presence of `[project]` section (standard PEP 621 project)
+4. **Unknown**: If none of the above sections are found
+
+### Priority System
+
+While a `pyproject.toml` file can technically contain configuration for multiple build systems, in practice, developers typically choose one primary build system per project. 
+
+When multiple build system configurations are detected in a single `pyproject.toml` file, the CLI uses the following priority order:
+
+1. Poetry (highest priority)
+2. PDM
+3. PEP 621
+4. Unknown (lowest priority)
+
+This prioritization ensures we select the most likely active build system when multiple configurations are present, avoiding confusion and duplicate analysis results. 
```

Comment:
What makes Poetry most likely as opposed to the others? 

The [spec](https://packaging.python.org/en/latest/specifications/pyproject-toml/#declaring-build-system-dependencies-the-build-system-table) only says that these are the lists of dependencies needed to build the project. Pip also [describes](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/#build-time-dependencies) it this way. 

I think that using this table value as a way to identify the underlying build system is actually the wrong approach. It'd be better to search for the lock file for the respective build system.